### PR TITLE
[Documentation] PSR12 - Return Type Declaration

### PR DESCRIPTION
**Pull request by dingo-d**
The PR contains the documentation for the PSR12/Functions/ReturnTypeDeclaration sniff.

## Description
This PR will add the documentation for the above-mentioned sniff, according to the [official standard definitions](https://www.php-fig.org/psr/psr-12/#45-method-and-function-arguments).

## Suggested changelog entry
Add documentation for the PSR12 ReturnTypeDeclaration sniff

## Related issues/external references

Part of #148

## Types of changes

- [ ] Bug fix _(non-breaking change which fixes an issue)_
- [ ] New feature _(non-breaking change which adds functionality)_
- [ ] Breaking change _(fix or feature that would cause existing functionality to change)_
    - [ ] This change is only breaking for integrators, not for external standards or end-users.
- [x] Documentation improvement

## PR checklist

- [x] I have checked there is no other PR open for the same change.
- [x] I have read the [Contribution Guidelines](https://github.com/PHPCSStandards/PHP_CodeSniffer/blob/master/.github/CONTRIBUTING.md).
- [x] I grant the project the right to include and distribute the code under the BSD-3-Clause license (and I have the right to grant these rights).
- [ ] I have added tests to cover my changes.
- [x] I have verified that the code complies with the projects coding standards.
- [ ] [Required for new sniffs] I have added XML documentation for the sniff.

### CODE REVIEW

File: src/Standards/PSR12/Docs/Functions/ReturnTypeDeclarationStandard.xml [hunk 1]
```diff
@@ -0,0 +1,39 @@
+<documentation title="Return Type Declaration">
+    <standard>
+    <![CDATA[
+    For functions and closures, the return type declaration must be correctly defined.
```
Comment (jrfnl) [inline src/Standards/PSR12/Docs/Functions/ReturnTypeDeclarationStandard.xml:4]: As a standards description, this feels quite _non-descript_.

What about using the [PSR12 phrasing](https://www.php-fig.org/psr/psr-12/#45-method-and-function-arguments) (phrase just above the third code sample in the section) ?

**Review by jrfnl — COMMENTED** [review]
Thanks for this PR @dingo-d !

Just one remark about the description. Everything else seems perfectly fine.

File: src/Standards/PSR12/Docs/Functions/ReturnTypeDeclarationStandard.xml (hunk 1 above)
Comment (dingo-d) [reply src/Standards/PSR12/Docs/Functions/ReturnTypeDeclarationStandard.xml:4]: I will update the description 👍🏼 This does feel a bit vague 😅

**Review by jrfnl — APPROVED** [review]
Thanks for updating the description @dingo-d ! LGTM.

**Comment by jrfnl** [comment]
Rebased and squashed, merging once the build passes.

**Pull request merged**